vignettes/compareToBaseR.Rmd

```diff
@@ -0,0 +1,683 @@
+
+```{r}
+fruit <- c("apple", "banana", "pear", "pinapple")
+
+## base
+grepl(pattern = "a", x = fruit)
+
+## stringr
+str_detect(string = fruit, pattern = "a")
+```
+
+In base you would use `grepl()` (see the "l" and think logical) while in stringr you use `str_detect()` (see the verb "detect" and think of a yes/no action). Note the different order of inputs. `grepl()` expects the pattern of interest first while `str_detect()` expects the string first. This order of inputs can be useful when combining `str_detect()` with `apply()` or `purrr::map()` statements.
+
+
+## `str_which`
+
+*Find positions matching a pattern*
+
+Now you want to identify the positions of the words in a vector of fruit names that contain an "a". 
+
+
+```{r}
+## base
+grep(pattern = "a", x = fruit)
+
+## stringr
+str_which(string = fruit, pattern = "a")
+```
+
+In base you would use `grep()` while in stringr you use `str_which()` (analogous to `which()` returning matching positions). Again there is a difference in the order of expected inputs. It should be noted that `grep()` is a versatile function. You will see it used again in this vignette for a different purpose.
+
+
+## `str_count`
+
+*Count the number of matches in a string.*
+
+How many "a"s are in each fruit?
+
+```{r}
+## base 
+gregexpr(pattern = "a", text = fruit, fixed = TRUE)
+
+
+## stringr
+str_count(string = fruit, pattern = "a")
+```
+
+This information can be gleaned from `gregexpr()` in base, but you will need to parse the output. This functionality is much more straightforward in stringr's `str_count()`. 
+
+
+## `str_locate`
+
+*Locate the position of patterns in a string.*
+
+Within each fruit, where does the first "p" occur? all of the "p"s?
+
+```{r}
+## base
+gregexpr(pattern = "p", text = fruit)
+
+## stringr
+str_locate(string = fruit, pattern = "p")
+str_locate_all(string = fruit, pattern = "p")
+```
+
+Again we would need to parse the output of `gregexpr()` to get the information we want. The stringr package provides specialized functions, `str_locate()` and `str_locate_all()` to do the job. 
+
+
+
+# Subset strings
+
+## `str_sub`
+
+*Extract and replace substrings from a character vector*
+
+What if we want to grab part of a string?
+
+```{r}
+hw <- "Hadley Wickham"
+
+## base
+substr(x = hw, start = 1, stop = 6)
+substring(text = hw , first = 1) 
+
+## stringr
+str_sub(string = hw, start = 1, end = 6)
+str_sub(string = hw, start = 1)
+str_sub(string = hw, end = 6)
+```
+
+In base you could use `substr()` or `substring()`. The former requires both a start and stop of the substring while the latter assumes the stop will be the end of the string. The stringr version, `str_sub()` has the same functionality, but also gives a default start value (the beginning of the string). Both the base and stringr functions have the same order of expected inputs. 
+
+
+There are some differences in behavior between `str_sub()` and `substr()`.
```

Comment:
I think this should be in a separate section, since this is technically a different function `str_sub<-` — I'd call this subset assignment or similar.